modules/saslplainauth.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2004-2025 ZNC, see the NOTICE file for details.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <znc/User.h>
+#include <znc/znc.h>
+
+class CSASLMechanismPlain : public CModule {
+  public:
+    MODCONSTRUCTOR(CSASLMechanismPlain) { AddHelpCommand(); }
+
+    void OnClientGetSASLMechanisms(SCString& ssMechanisms) override {
+        ssMechanisms.insert("PLAIN");
+    }
+
+    EModRet OnClientSASLAuthenticate(const CString& sMechanism,
+                                     const CString& sMessage) override {
+        if (!sMechanism.Equals("PLAIN")) {
+            return CONTINUE;
+        }
+
+        CString sNullSeparator = std::string("\0", 1);
+        CString sAuthzId = sMessage.Token(0, false, sNullSeparator, true);
+        CString sAuthcId = sMessage.Token(1, false, sNullSeparator, true);
+        CString sPassword = sMessage.Token(2, false, sNullSeparator, true);
+
+        if (!sAuthzId.empty() && sAuthzId != sAuthcId) {
+            // Reject custom SASL plain authorization identifiers
+            GetClient()->RefuseSASLLogin("No support for custom AuthzId");
+            return HALTMODS;
+        }
+
+        GetClient()->StartSASLPasswordCheck(sAuthcId, sPassword);
+        return HALTMODS;
+    }
+};
+
+template <>
+void TModInfo<CSASLMechanismPlain>(CModInfo& Info) {
+    Info.SetWikiPage("saslplain");
```

Comment:
Forgotten in af2175390160b1558d844036b270d419b86471e9?

```suggestion
    Info.SetWikiPage("saslplainauth");
```